pkg/schemadsl/compiler/translator.go

```diff
@@ -103,6 +106,24 @@ func translate(tctx *translationContext, root *dslNode) (*CompiledSchema, error)
 
 	// Check for typechecking after all definitions are processed.
 	if slices.Contains(tctx.enabledFlags, "typechecking") {
+		resolver := ResolverForCompiledSchema(compiledSchema)
+		ts := schema.NewTypeSystem(resolver)
```

Comment:
This should be done as part of the `.Validate` call on the type system, rather than the schema translator

Reply:
one slight caveat there -- we lose the `tctx`. Now, an easy way to fix that is to strip the metadata if it's not enabled, and then check it in Validate, which I'll add as a commit